### PR TITLE
Japanese Keyboard - Optional Auto Switching

### PR DESCRIPTION
**Pull request by mzsanford**
I was curious if https://github.com/davidsansome/tsurukame/issues/221 was even possible given the iOS API. In researching it I fell into a bit of a hole and ended up implementing it to verify it works. I'm not sure if this is something worth including because as the discussion in the issue points out, the Japanese keyboard kanji suggestions end up being akin to cheating. I have been working on practicing the kana keyboard so I may take this new found knowledge and make a simple practice app. Even if this isn't merged it was a good learning exercise.

### CODE REVIEW

File: ios/ReviewViewController.swift [hunk 1]
```diff
@@ -542,6 +542,23 @@ class ReviewViewController: UIViewController, UITextFieldDelegate, TKMSubjectDel
...
     }
 
+    if Settings.autoSwitchKeyboard {
+      let answerLanguage: String
+      if kanaInput.enabled {
+        answerLanguage = "ja"
+      } else {
+        answerLanguage = "en"
+      }
+      if answerField.answerLanguage != answerLanguage {
```
Comment (davidsansome) [inline ios/ReviewViewController.swift]: This logic would probably belong better inside AnswerTextField in answerLanguage's setter.  Then the code here could just be:

    answerField.answerLanguage = kanaInput.enabled ? "ja" : "en"

**Comment by turtlemaster19** [comment]
This is great news! @mzsanford, can you tell me how you get it to build on your computer? I would like to be able to build the repo, but I think I missed How To Build Someone Else's iOS Source 101. Thank you! 😉

**Comment by mzsanford** [comment]
@turtlemaster19 I started a chat at https://gitter.im/tsurukame and put the initial instructions there (the `community` room). I used that because I expect there will be some steps I don't remember. Once you've tested the steps and it's working I'll submit a PR to add it to the README.

**Comment by turtlemaster19** [comment]
I got that far. After I tried building, I got this:
13Errors

**Comment by mzsanford** [comment]
@turtlemaster19 you may need to run `pod install` from the `ios/` subdirectory to install the dependencies. If you don't have the `pod` command you can install it from https://cocoapods.org/.

**Comment by turtlemaster19** [comment]
@mzsanford The error messages changed after that. I didn't know about the `pod` command before, but I still have this going on after that:
13 build errors still

**Comment by turtlemaster19** [comment]
For example, I have the `Reachability.h` file at `ios/Pods/Reachability/Reachability.h` instead of the `ios` directory

**Comment by osdiab** [comment]
How do I use this in app? I’ve been wanting this for a while but I don’t see an option?

**Comment by osdiab** [comment]
Oh is this not published yet?

**Comment by davidsansome** [comment]
Yeah this hasn't been published yet, sorry!  It'll be in the next release!

**Comment by osdiab** [comment]
great! looking forward to it. Any details on when that might be?

**Comment by turtlemaster19** [comment]
@osdiab You can sign up for Matt Sanford's beta now: there's a link [here](https://community.wanikani.com/t/ios-tsurukame-native-app-with-offline-lessons-and-reviews/30635/456).
You can also build it from source yourself.
Or, of course, you could wait for David Sansome to initiate the review process with the App Store.

**Comment by osdiab** [comment]
Great to know about the TestFlight :)

**Pull request merged**